include/xtensor/xview.hpp

```diff
@@ -1650,7 +1658,14 @@ namespace xt
     inline auto xview<CT, S...>::sliced_access(const xslice<T>& slice, Arg arg, Args... args) const -> size_type
     {
         using ST = typename T::size_type;
-        return static_cast<size_type>(slice.derived_cast()(argument<I>(static_cast<ST>(arg), static_cast<ST>(args)...)));
+        if constexpr (I < sizeof...(Args)+1) {
+            return static_cast<size_type>(slice.derived_cast()(argument<I>(static_cast<ST>(arg), static_cast<ST>(args)...)));
+        }
+        else {
```

Comment:
**Possible fix**: I moved the static_assert here and produce a runtime exception if the code is really called.
(This code can be back ported to C++14 - but it is more readable as C++17 code for our discussion.)

All unit tests (`make xtest`) pass (including the new tests from this PR).

**Any comments are welcome to discuss this possible code change**. I see no performance impact, since the decision is still at compile time - only that it compiles and reports the error only at runtime (if called).

**Note:** The real reason is yet not understood. Only the effect is controlled.

Reply:
I back ported the C++ 17 code to C++ 14 code to allow all employed compilers in the CI process to compile the code and run all tests including the new tests (successfully).